### DISCUSSION TITLE
How top programmatically open files?

### DISCUSSION THREAD

**Discussion by treagod** [elementary/code#1158] [Q&A]
Hi I'm trying to add a fuzzy search ([see my branch here](https://github.com/treagod/code/tree/fuzzy-finder)) and the concept is really working well for now. 

I'm now at the point where I want to open a file which was selected by the fuzzy finder ([see](https://github.com/treagod/code/blob/bf376c8ba718f7b613a6919e3ca5239c944d36f1/plugins/fuzzy-search/fuzzy-search.vala#L80) `plugins/fuzzy-search/fuzzy-search.vala line 80`). But `open_file` from the interface can't open the file because it says the file is corrupt (it isn't!).
If the selected file is already open the editor will crash.

Can somebody guide me here how to programmatically open a file from a plugin? I'm kind of lost.

I'm getting following segmentation fault when opening the file:
![open_file_error](https://user-images.githubusercontent.com/4701538/146010760-fdd64afd-8372-4d3e-a5c9-3713cde28b61.png)

**Comment by jeremypw** [comment]
Plugins have access to the MainWindow so you could try creating a Services.Document from the file and then call the `open_document` function.

**Reply by treagod** [reply]
Thanks for the quick reply.

I tried it, but I'm still getting the same error:
![editor_error_window](https://user-images.githubusercontent.com/4701538/146049127-8fedf611-6848-4697-9182-ff1a8ac06889.png)

As far as I can tell the plugin interface method `open_file` is doing this internally.

**Reply by jeremypw** [reply]
Looks like you have a `null` file?

**Comment by jeremypw** [comment]
If you make and submit a draft PR it will be easier to investigate.

**Reply by treagod** [reply]
Created #1159
